controller/controller.go

```diff
@@ -0,0 +1,98 @@
+// Package controller implements a controller client providing basic operations against a
+// controller endpoint through controller's HTTP API. Higher level scheduling and
+// management should be built on top of that.
+package controller
+
+import (
+	"github.com/giantswarm/formica/fleet"
+)
+
+type Config struct {
+	Fleet fleet.Fleet
+}
+
+func DefaultConfig() Config {
+	newFleetConfig := fleet.DefaultConfig()
+	newFleet, err := fleet.NewFleet(newFleetConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	newConfig := Config{
+		Fleet: newFleet,
+	}
+
+	return newConfig
+}
+
+// Controller defines the interface a controller needs to implement to provide
+// operations for groups of unit files against a fleet cluster.
+type Controller interface {
+	// Submit schedules a group on the configured fleet cluster. This is done by
+	// setting the units target state to loaded.
```

Comment:
"This is done by setting the target state of the units in the group to `loaded`" :flag-gb:, same below as well